staging/src/k8s.io/api/core/v1/field_constants.go

```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package core
+package v1
 
 // Field path constants that are specific to the internal API
 // representation.
```

Comment:
Taking the fields one by one:

```
	NodeUnschedulableField = "spec.unschedulable"
```
this is related to v1 Nodes, but is unreferenced. I would drop it rather than export it.

```
	ObjectNameField        = "metadata.name"
```
this is not only external but is cross-version... I could see it moving to `staging/src/k8s.io/apimachinery/pkg/apis/meta/v1`

```
	PodHostField           = "spec.nodeName"
	PodStatusField         = "status.phase"
```
these are related to v1 Pods, but we don't typically export field selector constants... what would it look like to use inline strings instead of exporting as an API constant?

```
	SecretTypeField        = "type"
```
this is related to v1 Secrets, but we don't typically export field selector constants... what would it look like to use inline strings instead of exporting as an API constant?

```
	EventReasonField                  = "action"
	EventSourceField                  = "reportingComponent"
	EventTypeField                    = "type"
	EventInvolvedKindField            = "involvedObject.kind"
	EventInvolvedNamespaceField       = "involvedObject.namespace"
	EventInvolvedNameField            = "involvedObject.name"
	EventInvolvedUIDField             = "involvedObject.uid"
	EventInvolvedAPIVersionField      = "involvedObject.apiVersion"
	EventInvolvedResourceVersionField = "involvedObject.resourceVersion"
	EventInvolvedFieldPathField       = "involvedObject.fieldPath"
```
These are not referenced anywhere, but match values used in core/v1 event field selectors https://github.com/kubernetes/kubernetes/blob/release-1.19/pkg/apis/core/v1/conversion.go#L426-L448) and internally (https://github.com/kubernetes/kubernetes/blob/release-1.19/pkg/registry/core/event/strategy.go#L111-L123)

The events.k8s.io API converts field names to these as well: 
* https://github.com/kubernetes/kubernetes/blob/release-1.19/pkg/apis/events/v1/conversion.go#L63-L77
* https://github.com/kubernetes/kubernetes/blob/release-1.19/pkg/apis/events/v1beta1/conversion.go#L63-L77

Reply:
We don't typically export constants for field selector keys. `ObjectNameField` is the only constant here I would expect us to export, and it seems like it should be in the meta/v1 package more than the core/v1 package.

Reply:
>  ObjectNameField        = "metadata.name"

I moved `ObjectNameField` to `staging/src/k8s.io/apimachinery/pkg/apis/meta/v1`


> PodHostField    = "spec.nodeName"
SecretTypeField = "type"

`PodHostField` has been widely used in `fields.OneTermEqualSelector(metav1.PodHostField, node.Name)`
`SecretTypeField`   will break the comment 

#https://github.com/kubernetes/kubernetes/blob/release-1.19/cmd/kubeadm/app/cmd/token.go#L373-L378
#https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/controller-manager/pkg/clientbuilder/client_builder.go#L43-L44

Are you sure to change them all?

Reply:
> `PodHostField` has been widely used in `fields.OneTermEqualSelector(metav1.PodHostField, node.Name)`

There's only a single use outside of test files. It's fine to change those from a constant to an inline field selector.

> `SecretTypeField` will break the comment
> 
> #https://github.com/kubernetes/kubernetes/blob/release-1.19/cmd/kubeadm/app/cmd/token.go#L373-L378
> #https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/controller-manager/pkg/clientbuilder/client_builder.go#L43-L44
> 
> Are you sure to change them all?

Yes, inlining the field selector is fine.